api/disk.go

```diff
@@ -0,0 +1,27 @@
+package api
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+)
+
+func (api *API) ResizeDisk(instanceID int, params map[string]interface{}) (map[string]interface{}, error) {
+	data := make(map[string]interface{})
+	failed := make(map[string]interface{})
+	id := strconv.Itoa(instanceID)
+	log.Printf("[DEBUG] go-api::disk::resize instance ID: %s", id)
+	path := fmt.Sprintf("api/instances/%s/disk", id)
+	response, err := api.sling.New().Put(path).BodyJSON(params).Receive(&data, &failed)
+	if err != nil {
+		return nil, err
+	}
+	if response.StatusCode != 200 {
+		return nil, fmt.Errorf("ResizeDisk failed, status: %v, message: %s", response.StatusCode, failed)
```

Comment:
Just realised, `failed` isn't a string, the body of error responses look like this

```json
{
  "error": "Disk space can only be changed every 8 hour.",
  "error_code": 40002
}
``` 

`Error: ResizeDisk failed, status: 400, message: map[error:Disk space can only be changed every 8 hour. error_code:%!s(float64=40002)]`

This is minor, we can update it later